server/api/api_accessor.py

```diff
@@ -3,39 +3,47 @@
 import asyncio
 
 import pkg_resources
-from httplib2 import Http
-from oauth2client.service_account import ServiceAccountCredentials
-from server.config import API_TOKEN_URI, API_BASE_URL
+from server.config import API_TOKEN_URI, API_BASE_URL, API_CLIENT_SECRET, API_CLIENT_ID
+from oauthlib.oauth2 import BackendApplicationClient
+from requests_oauthlib import OAuth2Session
+import os
+os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
 
 CACERTS_FILE = pkg_resources.resource_filename('static', 'cacerts.txt')
 
 
 class ApiAccessor:
-    def __init__(self):
-        self._service_account_credentials = ServiceAccountCredentials.from_p12_keyfile(
-            'faf-server',
-            'faf-server.pem',
-            scopes='write_achievements write_events'
-        )
-        self._service_account_credentials.token_uri = API_TOKEN_URI
-
-    async def api_get(self, path, player_id):
-        loop = asyncio.get_event_loop()
-        result = await loop.run_in_executor(None, partial(self.http(player_id).request, API_BASE_URL + path, "GET"))
-        return result
+    def __init__(self, backend_app_client=BackendApplicationClient, oauth_session=OAuth2Session):
+        client = backend_app_client(client_id=API_CLIENT_ID)
+        self._session = oauth_session(client=client)
+        self._token = self._session.fetch_token(token_url=API_TOKEN_URI, client_id=API_CLIENT_ID, client_secret=API_CLIENT_SECRET)
+
+    async def update_achievements(self, achievements_data, player_id):
+
+        # Converting the achievements to a format the jAPI can understand
+        for achievement in achievements_data:
+            achievement['playerId'] = player_id
+            achievement['achievementId'] = achievement.pop('achievement_id')
+            achievement['operation'] = achievement.pop('update_type')
+
+        code, text = await self.api_patch("achievements/update", achievements_data)
+        return code, text
+
+    async def update_events(self, events_data, player_id):
+
+        # Converting the events to a format the jAPI can understand
+        for event in events_data:
+            event['playerId'] = player_id
+            event['eventId'] = event.pop('event_id')
 
-    async def api_post(self, path, player_id, data=None, headers=None):
-        headers = headers or {'Content-type': 'application/json'}
-        loop = asyncio.get_event_loop()
-        result = await loop.run_in_executor(
-            None,
-            partial(self.http(player_id).request, API_BASE_URL + path, "POST", headers=headers, body=json.dumps(data))
-        )
+        code, text = await self.api_patch("events/update", events_data)
+        return code, text
 
+    async def api_get(self, path):
+        result = self._session.get(API_BASE_URL + path)
```

Comment:
Why, `api_get` returns full result, but `api_patch`, `update_events` and `update_achievements` tuple?

Reply:
Fixed